atomic_reactor/plugins/exit_delete_from_registry.py

```diff
@@ -42,9 +42,38 @@ def __init__(self, tasker, workflow, registries):
 
         self.registries = deepcopy(registries)
 
+    def check_for_orchestrator_build(self):
+        """
+         If we are being called from an orchestrator build, collect the worker
+         node data and recreate the data locally.
+        """
+        if are_plugins_in_order(self.workflow.buildstep_plugins_conf, 'orchestrate_build'):
+            # orchestrator node
+            if not self.workflow.build_result.annotations:
+                return
+
+            builds = self.workflow.build_result.annotations['worker-builds']
+
+            for plat, annotation in builds.items():
+                self.log.debug("build %s has annotation: %s" % (plat, annotation))
+                digests = annotation['all-digests']
+
+                for digest in digests:
+                    registry = None
+                    for reg in self.workflow.push_conf.docker_registries:
+                        if reg.uri == digest['registry']:
+                            registry = reg
+
+                    if not registry:
+                        registry = self.workflow.push_conf.add_docker_registry(digest['registry'])
```

Comment:
I thought that too at first, but actually it makes sense. In the orchestrator build, push_conf will only have the spec for the manifest list (or image manifest) it creates in the registry. By changing this here, we'll add to that the specs for the image manifests the worker builds created.

Then they are all deleted from the registry together.